pkg/base/replicated.go

```diff
@@ -152,7 +157,62 @@ func renderReplicated(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (
 		}
 	}
 
-	return &commonBase, helmBases, nil
+	return &commonBase, helmBases, renderedKotsKinds, nil
+}
+
+func renderKotsKinds(upstreamFiles []upstreamtypes.UpstreamFile, renderedConfig *kotsv1beta1.Config, renderOptions *RenderOptions, builder *template.Builder) (map[string][]byte, error) {
+	fileContentsMap := make(map[string][]byte)
+
+	for _, upstreamFile := range upstreamFiles {
+		gvk := OverlySimpleGVK{}
+		if err := yaml.Unmarshal(upstreamFile.Content, &gvk); err != nil {
```

Comment:
what if the file is a multi-yaml document that contains multiple kots kinds? when loading kots kinds, we use `util.ConvertToSingleDocs` to convert to single docs.

Reply:
good call, changed it to use the utility doc splitting function and loop over the results